### PR TITLE
Added more network requestors

### PR DESCRIPTION
**Pull request by evilpenguin**
### Added
- Full search
- Person search (name, city, country)
- Petition search (name, type)
- Donation search (name, type)

### CODE REVIEW

File: SayTheirNames/Source/Service/Service.swift [hunk 1]
```diff
@@ -42,5 +42,14 @@ final class Service: Servicing {
         Log.mode = .all
         Log.print("STN Version: \(Bundle.versionBuildString)")
         Log.print("Starting Services")
+        
+        self.network.fetchPeopleByName("george") { (result) in
```
Comment (Kilo-Loco) [inline SayTheirNames/Source/Service/Service.swift]: should we remove this?

Comment (evilpenguin) [reply SayTheirNames/Source/Service/Service.swift]: Yeah I pushed after you started the review.

File: SayTheirNames/Source/Model/Search.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import Foundation
+
+public struct SearchResponsePage: Decodable {
+    let people: [Person]
+    let donations: [Donation]
+    let petitions: [Petition]
+    let link: Link
```
[the remark this answers is not in the document]
Comment (evilpenguin) [reply SayTheirNames/Source/Model/Search.swift:31]: We will always have a `Link` object from the backend. The values will just be `nil` some cases.

**Comment by evilpenguin** [comment]
> Kickstarting workflow again...

Couple changes coming!

**Comment by evilpenguin** [comment]
> I'm wondering if we should be working with `URLQueryItem`s instead of using String interpolation to construct the urls

Added

**Pull request merged**